tools/mtouch/mtouch.cs

```diff
@@ -372,7 +372,12 @@ public static string GetAotCompiler (Application app, bool is64bits)
 					return Path.Combine (cross_prefix, "bin", "arm-darwin-mono-sgen");
 				}
 			case ApplePlatform.WatchOS:
-				return Path.Combine (cross_prefix, "bin", "armv7k-unknown-darwin-mono-sgen");
+				/* Use arm64_32 cross only for Debug mode */
+				if (app.IsArchEnabled (Abi.ARM64_32) && !app.EnableLLVMOnlyBitCode) {
```

Comment:
Ahh, thanks!  Right, because `MtouchArch` can be set to `ARMv7k, ARM64_32`. However `MtouchExtraArgs` cannot be set separately; `--interpreter` will not work for `ARMv7k`.

In terms of user experience, any suggestion what we should do here? Maybe do not require the user to set `--interpreter` explicitly in `MtouchExtraArgs`, but force it in `mtouch` when `ARM64_32` + `Debug Mode` is detected?

Reply:
> force it in `mtouch` when `ARM64_32` + `Debug Mode` is detected

I think this would be the best option for customers, but I think we'd want to emit a warning in this case explaining what we're doing (I don't like silently changing behavior), and explain that it's possible to get the normal AOT compiler by disabling Debug mode in the Debug configuration (I'm assuming that actually works).

That said, this won't be a problem if customers have device-specific builds turned on (which iirc is on by default in new projects).

Reply:
"normal" AOT aka. FullAOT for `arm64_32` doesn't work currently, hence the interpreter. It would only be needed for `Debug Mode`, but since we can use the interpreter there, I'm not sure if we will ever need to do the work to support FullAOT.

Reply:
> "normal" AOT aka.

Right now people can use Debug mode and it works, except that you can't debug. My concern is that if someone runs into a bug in the interpreter, they should be able to stop using the interpreter somehow so that the app runs correctly on the watch, even though debugging doesn't work (they could at least use Console.WriteLines).

Reply:
No, FullAOT does not work at all for `arm64_32` currently 😕